qualtran/bloqs/arithmetic/Error propagation due to fixed point arithmetic.ipynb

```diff
@@ -0,0 +1,203 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "6d12f4fc-8129-4c12-a00f-fe52aa4888e5",
+   "metadata": {},
+   "source": [
+    "## Fixed point arithmetic error propagation analysis using fxpmath\n",
+    "\n",
+    "In this notebook, we use library fxpmath to analyze the propagation of errors due to fixed point arithmetic for algorithms described in [Compilation of Fault-Tolerant Quantum Heuristics for Combinatorial Optimization](https://arxiv.org/abs/2007.07391). \n",
+    "\n",
+    "Qualtran should provide tools to enable such analysis for algorithms that use fixed point arithmetic Bloqs as subroutines. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "92cdfbaa-1e5f-4aac-9fdd-4d4015fd384a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from fxpmath import Fxp\n",
+    "import numpy as np\n",
+    "\n",
+    "# Initial setup - Helpers\n",
+    "def fxp(x: float, d: int) -> Fxp:\n",
+    "    dtype = f'fxp-u{d}/{d}'\n",
```

Comment:
docstring is necessary here, esp if one is not familiar with fxpmath

Reply:
added docstrings